src/common/gui/SurgeGUIEditor.cpp

```diff
@@ -2127,11 +2128,35 @@ void SurgeGUIEditor::valueChanged(CControl* control)
       // frame->setModalView(nullptr);
       frame->setDirty();
 
-      synth->storage.getPatch().name = patchName->getText();
-      synth->storage.getPatch().author = patchCreator->getText();
-      synth->storage.getPatch().category = patchCategory->getText();
-      synth->storage.getPatch().comment = patchComment->getText();
-      synth->savePatch();
+      /*
+      ** Don't allow a blank patch
+      */
+      std::string whatIsBlank = "";
+      bool haveBlanks = false;
+
+      if (std::string(patchName->getText()).find_first_not_of( ' ' ) == std::string::npos)
+      {
+          whatIsBlank = "name"; haveBlanks = true;
+      }
+      if (std::string(patchCategory->getText()).find_first_not_of( ' ' ) == std::string::npos)
```

Comment:
`validatePatchName()` could be called in both of these call sites. 

I would also use it when we scanning all the patches in SurgeStorage, but that is up to you. I would just skip the patches that have invalid filenames, possibly giving one promptError() after scanning them that reports the number of patches that were skip because of this reason. Allowing to load such patches brings it own set of corner cases, as you can probably understand.